vpr/src/place/place_constraints.cpp

```diff
@@ -413,3 +413,53 @@ bool is_pr_size_one(PartitionRegion& pr, t_logical_block_type_ptr block_type, t_
 
     return pr_size_one;
 }
+
+int get_region_size(const Region& reg, t_logical_block_type_ptr block_type) {
```

Comment:
Could I also just use a 2d matrix where at each x, y location, I store the number of subtiles that are compatible with the block - instead of adding the third dimension of that goes up to max_subtiles?

Reply:
You could do that, but it will be an overestimate for the case where a specific sub-tile is requested.  You could comment that behaviour (or probably better: drop back to slower code that goes through the whole region in that case as a large region and a specific subtile would be unusual).  So if you prefer to code it that way  I'm OK with it.